easybuild/main.py

```diff
@@ -586,9 +481,198 @@ def main(args=None, logfile=None, do_build=None, testing=False, modtool=None):
+        options.create_index,
+        options.install_github_token,
+        options.list_installed_software,
+        options.list_software,
+        options.close_pr,
+        options.list_prs,
+        options.merge_pr,
+        options.review_pr,
+        options.terse,
+        search_query,
+    ]
+    if any(early_stop_options):
+        clean_exit(logfile, eb_tmpdir, testing, silent=True)
+
+    # update session state
+    eb_config = eb_go.generate_cmd_line(add_default=True)
+    modlist = modtool.list()  # build options must be initialized first before 'module list' works
+    init_session_state.update({'easybuild_configuration': eb_config})
+    init_session_state.update({'module_list': modlist})
+    _log.debug("Initial session state: %s" % init_session_state)
+
+    if options.skip_test_step:
+        if options.ignore_test_failure:
+            raise EasyBuildError("Found both ignore-test-failure and skip-test-step enabled. "
+                                 "Please use only one of them.")
+        else:
+            print_warning("Will not run the test step as requested via skip-test-step. "
+                          "Consider using ignore-test-failure instead and verify the results afterwards")
+
+    # determine easybuild-easyconfigs package install path
+    easyconfigs_pkg_paths = get_paths_for(subdir=EASYCONFIGS_PKG_SUBDIR)
+    if not easyconfigs_pkg_paths:
+        _log.warning("Failed to determine install path for easybuild-easyconfigs package.")
+
+    if options.install_latest_eb_release:
+        if orig_paths:
+            raise EasyBuildError("Installing the latest EasyBuild release can not be combined with installing "
+                                 "other easyconfigs")
+        else:
+            eb_file = find_easybuild_easyconfig()
+            orig_paths.append(eb_file)
+
+    if options.easystack:
+        _log.debug("Start build loop over items in the EasyStack file: %s" % orig_paths)
+        # TODO: insert fast loop that validates if all command line options are valid. If there are errors in options,
+        # we want to know early on, and this loop potentially builds a lot of packages and could take very long
+        # for path in orig_paths:
+        #     validate_command_opts(args, opts_per_ec[path])
+        # Loop over each item in the EasyStack file, each time updating the config
+        # This is because each item in an EasyStack file can have options associated with it
+        for path in orig_paths:
+            _log.debug("Starting build for %s" % path)
+            # NOTE: not sure if this is needed. Is the EasyConfigs cache preserved throughout loops of this iteration?
+            # Current 'path' may have different options associated with it. Thus, resolution of EasyConfigs
+            # to full paths should not be read from cache, but redetermined. Thus, we wipe the cache
+            # wipe_cache()  # Wipes the easyconfig cache with _easyconfigs_cache.clear()  # TODO: implement
```

Comment:
Which of https://github.com/easybuilders/easybuild-framework/blob/d7409b3d68516a0bf70688d14e5750dedfbb8758/test/framework/utilities.py#L453 should it empty? All? Only the `_easyconfigs_cache`?

Reply:
It basically boils down to this I think (wipe both the cache of `EasyConfig` instances + the cache of where easyconfig files are located (see `robot_find_easyconfig` function)

```python
from easybuild.framework.easyconfig import easyconfig
...
easyconfig._easyconfigs_cache.clear()
easyconfig._easyconfig_files_cache.clear()
```

Reply:
fixed in 819138d1d532bf185ba2b63d880e8dfc1fb70ee6